citesphere/src/main/java/edu/asu/diging/citesphere/config/CustomAuthenticationFailureHandler.java

```diff
@@ -0,0 +1,59 @@
+package edu.asu.diging.citesphere.config;
+
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.DefaultRedirectStrategy;
+import org.springframework.security.web.RedirectStrategy;
+import org.springframework.security.web.authentication.AuthenticationFailureHandler;
+import org.springframework.security.web.util.UrlUtils;
+import org.springframework.util.Assert;
+
+public class CustomAuthenticationFailureHandler implements AuthenticationFailureHandler {
+
+	private final Logger logger = LoggerFactory.getLogger(getClass());
+	private String defaultFailureUrl;
+	private RedirectStrategy redirectStrategy = new DefaultRedirectStrategy();
+
+	public CustomAuthenticationFailureHandler() {
+	}
+
+	public CustomAuthenticationFailureHandler(String defaultFailureUrl) {
+		setDefaultFailureUrl(defaultFailureUrl);
+	}
+
+	@Override
+	public void onAuthenticationFailure(HttpServletRequest request, HttpServletResponse response,
+			AuthenticationException exception) throws IOException, ServletException {
+		
+		String errorMessage;
+		
+		if(exception.getMessage().equals("User account is locked")) {
+			errorMessage = "Your account has not been approved by an administrator yet.";
+		}else {
+			errorMessage = "Password or username are incorrect.";
+		}
```

Comment:
this is very error prone (what if they just change one letter in the exception message)? Instead, you should check what type of exception you are getting, then setting an error key. then use the messages file with that key for the actual error messages.

Reply:
otherwise, the error messages can never be localized if we ever plan to do that.

Reply:
" then setting an error key. then use the messages file with that key for the actual error messages.", not addressed?

Reply:
Oh yeah....I'll fix this.

Reply:
indentation